### PR TITLE
k8srollout: first pass at a pod rollout monitor

### PR DESCRIPTION
**Pull request by nicks**
Hello @hyu, @maiamcc,

Please review the following commits I made in branch nicks/pods:

7bc6c2d2613712b52d09ea1e9973a7b436871eed (2020-01-28 20:31:13 -0500)
k8srollout: first pass at a pod rollout monitor

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

**Comment by nicks** [comment]
here's a screenshot on success:

Screenshot from 2020-01-28 20-30-55

here's a screenshot while it's pending:
Screenshot from 2020-01-28 20-35-42

i have some ideas on how to make it visually pop out a bit more as it updates, tho this seems like a good base to iterate on

File: internal/engine/k8srollout/podmonitor.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+		m.trackingStarted[update.podID] = true
+	}
+
+	m.printCondition(ctx, "Scheduled", update.scheduled, update.startTime)
+	m.printCondition(ctx, "Initialized", update.initialized, update.scheduled.LastTransitionTime.Time)
+	m.printCondition(ctx, "Ready", update.ready, update.initialized.LastTransitionTime.Time)
+}
+
+func (m *PodMonitor) printCondition(ctx context.Context, name string, cond v1.PodCondition, startTime time.Time) {
+	l := logger.Get(ctx).WithFields(logger.Fields{logger.FieldNameProgressID: name})
```
Comment (maiamcc) [inline internal/engine/k8srollout/podmonitor.go]: am i right in thinking that this field updates in place?

Comment (nicks) [reply internal/engine/k8srollout/podmonitor.go]: yep! this is the api for printing progress

**Comment by nicks** [comment]
added a test

**Pull request merged**